### ISSUE TITLE
Boot path doesn't account for non-standard install locations

### ISSUE THREAD

**Issue by acfoltzer** [cisco/ChezScheme#6]
First off, 🎉 ! I'm so excited to see this repo and this license.

Once I silenced warnings just enough to get built, I ran into the issue when using a non-standard install path:

```
% ./configure --installprefix=/usr/local/ChezScheme
% make install
...
% /usr/local/ChezScheme/bin/scheme
% /usr/local/ChezScheme/bin/scheme
cannot find compatible scheme.boot in search path
  "/Users/acfoltzer/lib/csv%v/%m:/usr/lib/csv%v/%m:/usr/local/lib/csv%v/%m"
```

If I specify the boot path manually, the compiler starts fine.

**Comment by cathartes** [comment]
I ran into the same problem trying to install to /usr/lib64 on my slackware system.  The search path referred to in the output is hard-coded in **c/scheme.c** in the definition _defaultsystemheappath_ , so try changing that and doing the build again.  I also made adjustments to **configure** and **makefiles/Mf-install.in** on my system so I didn't have to keep re-tweaking my build files after running configure.

**Comment by rain-1** [comment]
This is the same issue as https://github.com/cisco/ChezScheme/issues/16

**Comment by cathartes** [comment]
rain-1, which did not exist at the time we posted.  ;-)

**Comment by rain-1** [comment]
@cathartes, would you be interested in sharing your fixes as a pull request?

**Comment by cathartes** [comment]
My own fixes are just duct tape. Ideally, **scheme.c** should reference a_defaultsystemheappath_ generated for the platform by **configure**.  I can submit a few tweaks, but I've only had a few hours to look at the source tree--someone with a better understanding of the build system would do a better job.  If someone (e.g., dybvig) doesn't beat me to it, I'll see what I can do!

**Comment by rain-1** [comment]
looking into this a little, is it really issue given the `SCHEMEHEAPDIRS` variable?

**Comment by cathartes** [comment]
As a stopgap, it's fine.  But, in my opinion, that variable is for overriding a normal (and working) default for a special use case, not for just getting the scheme to run.

**Comment by combinatorylogic** [comment]
Is this approach acceptable? https://github.com/cisco/ChezScheme/pull/25

**Comment by rain-1** [comment]
I applied your patch to my openbsd repo and used:

`./configure --nogzip-man-pages --installman=`pwd`/out --installprefix=`pwd`/out`

and make, make install worked great as a regular user. scheme finds the boot files too!

I don't know why the gzip-man-pages stage is failing though.

Looks great!

**Comment by cathartes** [comment]
Thank you, @combinatorylogic.  I put a pull request on a fork of your changes, with hope of maybe solving the man page issue that @rain-1 mentioned.  Seems to work well on my system, hopefully (both of) yours?

**Comment by cathartes** [comment]
I'm guessing the changes didn't meet standards, since the PR is just sitting there.  I'm probably to blame for that because of my messy commit messages.  Sorry!!

**Comment by akeep** [comment]
@cathartes I can't speak for everyone else, but I've just not had a chance to look through and test the changes in your PR in detail yet.

**Comment by akeep** [comment]
@cathartes Kent and I went through the PR #25.  We ended up making a few additional changes.

1. Changed the name of the header file to config.h (it is now entirely generated from the configure script)
2. Added --installbin, --installschemename, --installpetitename, and --installscriptname to try to complete our configuration options.
3. A variety of smaller, related changes to try to make some of these configuration changes work a bit smoother.

See the comments on df4194c for more details.

Thanks again for the PR!
